src/backend/gporca/libgpopt/src/xforms/CXformEagerAgg.cpp

```diff
@@ -261,6 +264,29 @@ CXformEagerAgg::CanApplyTransform
 		return false;
 	}
 
+    if (NULL == gb_agg_expr->Pstats())
+    {
+            CExpressionHandle exprhdl(mp);
+            exprhdl.Attach(gb_agg_expr);
+            exprhdl.DeriveStats(mp, mp, NULL /*prprel*/, NULL /*pdrgpstatCtxt*/);
+    }
+
+    if (NULL == join_expr->Pstats())
+    {
+            CExpressionHandle exprhdl(mp);
+            exprhdl.Attach(join_expr);
+            exprhdl.DeriveStats(mp, mp, NULL /*prprel*/, NULL /*pdrgpstatCtxt*/);
+    }
+
+    CDouble gbrows = gb_agg_expr->Pstats()->Rows();
+    CDouble joinrows = join_expr->Pstats()->Rows();
+	COptimizerConfig *optconfig = COptCtxt::PoctxtFromTLS()->GetOptimizerConfig();
+
+    if ((gbrows / joinrows) >= optconfig->GetHint()->EagerAggThreshold())
```

Comment:
The threshold has an allowed range between 0 and 1. It would be nice if a value of 1 always enables eager agg and 0 always disables it. Also, could joinrows ever be 0? It might be safer to use multiplication instead of division.

```suggestion
    CDouble threshold = optconfig->GetHint()->EagerAggThreshold();
    
    if ((gbrows > joinrows * threshold && threshold < 1) || threshold == 0)
```

Reply:
Makes sense

Reply:
+1 to Hans's suggestion. Just that I would reorder the if clause, and add comments (for readability):
```
CDouble threshold = optconfig->GetHint()->EagerAggThreshold();
// Always consider EagerAgg if threshold=0, but never when threshold=0
if (threshold == 0 || (gbrows > joinrows * threshold && threshold < 1))
```
Or you can split it (the comments aren't needed here really):
```
CDouble threshold = optconfig->GetHint()->EagerAggThreshold();
if (threshold >= 1)
{
    // always consider EagerAgg if threshold is 1
    return true;
}
else if (threshold == 0 || (gbrows > joinrows * threshold))
{
    // do not consider EagerAgg when threshold is 0 or if threshold is not met.
   return false;
}
```